source/common/upstream/upstream_impl.cc

```diff
@@ -2095,7 +2095,24 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
 
       // If we are depending on a health checker, we initialize to unhealthy.
       if (health_checker_ != nullptr && !host->disableActiveHealthCheck()) {
-        host->healthFlagSet(Host::HealthFlag::FAILED_ACTIVE_HC);
+        if (Runtime::runtimeFeatureEnabled(
+                "envoy.reloadable_features.keep_endpoint_active_hc_status")) {
+          if (existing_host_found && !health_check_address_changed &&
+              !active_health_check_flag_changed) {
```

Comment:
nit: this is only triggered when locality is changed right now, should we make the condition or the runtime guard naming explicit here?

Reply:
That's a good point.
My first take when fixing this was to keep the health-status regardless of whether the HC-address changed or the flag has changed.
If we keep the current conditions, then I agree the runtime flag name should be changed. Would like to hear others thoughts before making this change.

Reply:
+1 on more precise naming

Reply:
Done.